multitenancy-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/multitenancy/autoconfig/MultiTenantPersistentStreamAutoConfiguration.java

```diff
@@ -0,0 +1,69 @@
+package org.axonframework.extensions.multitenancy.autoconfig;
+
+import org.axonframework.axonserver.connector.event.axon.PersistentStreamMessageSource;
+import org.axonframework.axonserver.connector.event.axon.PersistentStreamMessageSourceFactory;
+import org.axonframework.common.StringUtils;
+import org.axonframework.extensions.multitenancy.components.TenantProvider;
+import org.axonframework.springboot.autoconfig.AxonServerAutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * Auto-configuration class for multi-tenant persistent stream support in Axon Framework.
+ * This configuration is enabled when Axon Server and multi-tenancy are both enabled.
+ *
+ * @author Stefan Dragisic
+ * @since 4.10
+ */
+@AutoConfiguration
+@ConditionalOnProperty(value = {"axon.axonserver.enabled", "axon.multi-tenancy.enabled"}, matchIfMissing = true)
+@AutoConfigureBefore(AxonServerAutoConfiguration.class)
+public class MultiTenantPersistentStreamAutoConfiguration {
+
+    /**
+     * Creates a PersistentStreamMessageSourceFactory for multi-tenant environments.
+     *
+     * @param tenantProvider The TenantProvider for managing tenants.
+     * @param tenantPersistentStreamMessageSourceFactory The factory for creating tenant-specific PersistentStreamMessageSources.
+     * @return A PersistentStreamMessageSourceFactory that supports multi-tenancy.
+     */
+    @Bean
+    @ConditionalOnMissingBean
+    public PersistentStreamMessageSourceFactory persistentStreamMessageSourceFactory(
```

Comment:
Can you help me understand why we have two factories right now? One for a `MultiTenantPersistentStreamMessageSource` and another to construct a `PersistentStreamMessageSource`?

Reply:
Aah yes, now it makes sense. Thanks for explaining, I think we're good here!